### PR TITLE
Refactor App to use Slugs

### LINKED ISSUE #805: Sidebar and breadcrumb links to services go to the dashboard
When viewing a specific instance of a service, the link at the top the sidebar with the service name and the breadcrumb with the service name (immediately to the right of the "fabric" breadcrumb) both navigate back to the dashboard page rather than to the expected service page (the one containing the list of instances).

### LINKED ISSUE #1437: Persistent Parenthetical Problem
## Actual Behavior

When a microservice has a parenthesis in the name, the instance dashboard renders an error condition when an instance is selected.

## Steps to Reproduce the Problem

  1.
  1.
  1.
  
## Specifications

  - Version:
  - Platform:
  - Subsystem:

### PR DESCRIPTION
**Pull request by spmcbride1201**
Note: Refactor currently breaks tons of tests. The mock JSON data needs to get refactored.

Resolves #805 
Resolves #1437 

## Proposed Changes

  - Generate slugs when we inject services from the discovery-service using the service name and version
  - Key services in Redux off the slug
  - Refactor Routers to collapse service name and version into this new slug
  
## How to Test

  - Run app and check for regressions
  - Run storybook and check for regressions

### CODE REVIEW

File: src/components/AppHeader/AppHeader.js [hunk 1]
```diff
@@ -27,18 +27,17 @@ function AppHeader({ location: { pathname } }) {
 }
 
 function getTitle(pathname) {
-  const [root, version = "", instance = ""] = decodeParameter(pathname)
+  const [selectedServiceSlug = "", selectedInstanaceID = ""] = pathname
```
Comment (kaitmore) [inline src/components/AppHeader/AppHeader.js]: typo here **selectedInstanceID

We should probably just connect AppHeader to state so we can lookup the real service name using the path. Maybe a follow up issue?

File: src/components/Main/scenes/FabricView/FabricRouter.js [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+              message = `${serviceSlug} does not have metrics to display`;
+              pathname = `/${serviceSlug}`;
+            }
+          }
+
+          // If the services object has not been passed to the router yet,
+          // or if the user is authorized and the service exists and is metered
+          // as determined by the message not getting set above
+          // then render the instance router, else redirect with error message
+          return servicesAreNotLoaded || !message ? (
```
Comment (kaitmore) [inline src/components/Main/scenes/FabricView/FabricRouter.js]: this is a lot better 🙌

File: src/components/Main/scenes/FabricView/FabricRouter.js [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+          // Set a message to pass to location state if one of the following checks fail
+          let message,
+            pathname = "/";
+
+          // Checks are ordered by priority of the message
+          // Check if the services are loaded, but the service slug didn't match
+          if (!servicesAreNotLoaded) {
+            if (!service) {
+              message = `${serviceSlug} is not a known microservice`;
```
Comment (kaitmore) [inline src/components/Main/scenes/FabricView/FabricRouter.js]: Probably should use just `service` for messages instead of `serviceSlug`

**Comment by kaitmore** [comment]
Awesome job!!! You've cleaned up an amazing amount of stuff here. 

I didn't notice any major regressions clicking through the app. The only thing is the service name in the header.

File: src/components/AppHeader/AppHeader.js (hunk 1 above)
Comment (spmcbride1201) [reply src/components/AppHeader/AppHeader.js]: Yeah, I agree that we should pull the actual service name. I'll open an issue for it.

**Pull request merged**